### PR TITLE
[FSI] New FSI coupling interface class

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR adds a new interface coupling class to ease the FSI coupling and relaxation. So far it is only used in the embedded FSI solver but eventually it will be used in the BF too.

### CODE REVIEW

File: applications/FSIapplication/python_scripts/fsi_coupling_interface.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+# Import applications
+import KratosMultiphysics.FSIApplication as KratosFSI
+
+class FSICouplingInterface():
+
+    def __ValidateSettings(self, settings):
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "model_part_name": "",
+            "father_model_part_name": "",
```
Comment (philbucher) [inline applications/FSIapplication/python_scripts/fsi_coupling_interface.py]:
```suggestion
            "parent_model_part_name": "",
```

please don't specify the gender xD

Joke aside, this is more coherent with the naming in ModelPart

File: applications/FSIapplication/tests/fsi_coupling_interface_test.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import KratosMultiphysics
+import KratosMultiphysics.KratosUnittest as UnitTest
+import KratosMultiphysics.FSIApplication as KratosFSI
+import KratosMultiphysics.StructuralMechanicsApplication as KratosStructural
+import KratosMultiphysics.kratos_utilities as KratosUtilities
+
+import os
+import sys
+import fsi_coupling_interface
```
Comment (philbucher) [inline applications/FSIapplication/tests/fsi_coupling_interface_test.py]:
```suggestion
from KratosMultiphysics.FSIApplication import fsi_coupling_interface
```

File: applications/FSIapplication/tests/fsi_coupling_interface_test.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import convergence_accelerator_factory
```
Comment (philbucher) [inline applications/FSIapplication/tests/fsi_coupling_interface_test.py]:
```suggestion
from KratosMultiphysics.FSIApplication import convergence_accelerator_factory
```

**Review by philbucher — DISMISSED** [review]
Some comments, otherwise looks good to me

In fact we have a similar concept in the CoSimApp

**Comment by philbucher** [comment]
I approved by mistake, but I can approve again if you take my suggestions into consideration

**Comment by philbucher** [comment]
also codacy has some suggestions

**Review by philbucher — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
Thanks @philbucher !

**Pull request merged**